### PR TITLE
Star 242 refractor css

### PR DESCRIPTION
**Pull request by KaanDarcey**
### Purpose 🚀
To resolve issue #98 by adjusting all current paprika components to follow the same standard styling patterns. 

No longer will styling be applied by css inline to the component, ex:

 ```<Component css={style} />```. 

Instead styling will be imported and used as follows, ex:

```
import * as sc from "./Card.styles";

  return (
    <sc.cardStyles {...cardProps} data-pka-anchor="card">
      {children}
    </sc.cardStyles>
  );
```

Additionally when using a prop functions inside of a css tag function it should destructure the props in the arrow function parameters ex:
```
export const cardStyles = styled(commonStyles)`
  ${props => {
    const size = sizeStyles[props.size];
    const active = props.isActive && activeStyles;
    const fullWidth = props.isFullWidth && fullWidthStyles;
    const autoHeight = props.isAutoHeight && autoHeightStyles;

    return `
      ${size};
      ${active};
      ${fullWidth};
      ${autoHeight};`;
  }}
`;
```
### Notes ✏️
- Removed  ```-webkit-appearance``` from Select component (Select.styles.js) as it conflicts with lint standards. Could be a problem for Safari browser.

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [ ] MINOR (backward compatible) change to _these packages_
- [x] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/STAR-242-refractor-css

### Screenshots 📸
N/A

### References 🔗
Issue #98

### CODE REVIEW

File: packages/Checkbox/src/Checkbox.styles.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ const styles = {
   },
 };
 
-const checkboxStyles = css`
+export const checkboxStyles = styled.div`
```
Comment (mikrotron) [inline packages/Checkbox/src/Checkbox.styles.js:78]: Since this is no longer just the styles, it's an actual component, I'd recommend we change the naming convention (this comment applies to all the components in this PR).  In this case, probably `CheckboxWrapper` (Note the capital `C`), since it's not the _actual_ checkbox.

File: packages/Counter/src/Counter.styles.js [hunk 2]
```diff
@@ -69,10 +69,17 @@ const counterStyles = css`
...
+    const hasIndicator = props.hasIndicator && indicatorStyles;
+    const color = counterColorStyles[props.color];
+    const size = counterSizeStyles[props.size];
+
+    return `
+      ${hasIndicator};
+      ${color};
+      ${size};`;
+  }}
+`;
```
Comment (mikrotron) [inline packages/Counter/src/Counter.styles.js:85]:
```suggestion
export const counterStyles = styled.span`
  ${commonStyles};
  ${props => `    
    ${props.hasIndicator && indicatorStyles}
    ${counterColorStyles[props.color]}
    ${counterSizeStyles[props.size]}
  `}
`;
```

Comment (mikrotron) [reply packages/Counter/src/Counter.styles.js:85]: Maybe that would be simpler? It assumes you remove `styled(commonStyles)` from line `74` too.

File: packages/FormElement/src/FormElement.styles.js [hunk 3]
```diff
@@ -16,10 +16,20 @@ const inlineFormElementStyles = css`
...
+    const size = FontSizes[props.size];
+    const isInline = props.isInline && inlineFormElementStyles;
+    const isDisabled = props.isDisabled && `opacity: 0.5;`;
+
+    return `
+    ${size};
+    ${isInline};
+    ${isDisabled};`;
+  }}
+`;
```
Comment (mikrotron) [inline packages/FormElement/src/FormElement.styles.js:35]: Why the change? I liked it better before 😕

**Comment by mikrotron** [comment]
This is a big chore, but thanks for taking it on! I didn't comment all the way through, but there are two recurring points I think should be addressed: 
1. The naming convention (the exports are now components, not styles, so they can be named like a component).
2. The `commonStyles` refactoring in several components has made the code longer and more indirect with no clear benefit that I can see.

File: packages/Checkbox/src/Checkbox.styles.js (hunk 1 above)
Comment (KDarcey) [reply packages/Checkbox/src/Checkbox.styles.js:78]: Will make sure to adjust all of the names where needed

File: packages/FormElement/src/FormElement.styles.js (hunk 3 above)
Comment (KDarcey) [reply packages/FormElement/src/FormElement.styles.js:35]: The primary reasoning was to maintain consistency with how I styled the other components, that being said I could understand why the previous version in this instance may be preferred as the code is certainly shorter. This is definitely something I could adjust if needed.

File: packages/Counter/src/Counter.styles.js (hunk 2 above)
Comment (KDarcey) [reply packages/Counter/src/Counter.styles.js:85]: I'll want to adjust the other components to follow the same styling pattern, but it shall be done 👍

File: packages/Counter/src/Counter.styles.js [hunk 4]
```diff
@@ -43,7 +43,7 @@ const counterColorStyles = {
   `,
 };
 
-const indicatorStyles = css`
+const indicatorStyles = `
```
Comment (mikrotron) [inline packages/Counter/src/Counter.styles.js:46]: I think it's a good idea to keep the `css` helper here, mainly because it will improve the syntax highlighting in our editors.

File: packages/Card/src/Card.js [hunk 5]
```diff
@@ -38,9 +38,9 @@ function Card(props) {
   };
 
   return (
-    <sc.cardStyles data-pka-anchor="card" {...styleProps} {...moreProps}>
+    <sc.CardWrapper data-pka-anchor="card" {...styleProps} {...moreProps}>
```
Comment (mikrotron) [inline packages/Card/src/Card.js:41]: I'm not a fan of the `Wrapper` suffix for naming in general – it's too vague to be helpful most of the time – but sometimes it really _is_ necessary.  But in this case I think simply `<sc.Card>` fits nicely, is a bit less typing, and a bit easier to read (will, at least when the component name also gets long).  

Bit of a hassle to make this adjustment throughout, but I think it's worth it to get it done right and then move on.  Thanks!

File: packages/ExternalLink/src/ExternalLink.styles.js [hunk 6]
```diff
@@ -3,21 +3,23 @@ import tokens from "@paprika/tokens";
...
+  ${props => `
   align-items: center;
   border-radius: ${tokens.border.radius};
   color: ${tokens.textColor.link};
   display: inline-flex;
   max-width: 100%;
   position: relative;
 
-  ${({ iconFontSize }) => `padding: 1px ${iconFontSize + toInt(tokens.spaceLg)}px 1px ${tokens.spaceSm};`}
+  ${`padding: 1px ${props.iconFontSize + toInt(tokens.spaceLg)}px 1px ${tokens.spaceSm};`}
```
Comment (mikrotron) [inline packages/ExternalLink/src/ExternalLink.styles.js:15]: I think when the prop is _only_ being used in this one spot it makes more sense _not_ to wrap the whole styling block in a function, but keep it scoped to the one line, as it was.  

Gets to be a bit of a judgement call then about when there's only _two_ lines... or _three_ lines... when to wrap the whole block... what do you think?

File: packages/FormElement/src/FormElement.styles.js [hunk 7]
```diff
@@ -8,18 +8,23 @@ const FontSizes = {
 };
 
 export const SectionsContainer = styled.div`
-  ${({ isInline }) => isInline && `flex-grow: 1;`}
+  ${props => `
```
Comment (mikrotron) [inline packages/FormElement/src/FormElement.styles.js:11]: I preferred the tidyness of the previous version where this could all be in one line.  I notice that you're consistently avoiding the destructuring of `props`, was there another reason besides consistency?  Was that approcah recommended by someone?

File: packages/Heading/src/Heading.styles.js [hunk 8]
```diff
@@ -52,17 +40,28 @@ const underline = level => {
...
+  color: inherit;
+  font-weight: 700;
+  letter-spacing: 0;
+  ${stylers.lineHeight(-1)};
+  margin: ${stylers.spacer(2)} 0;
+
+  &:focus {
+    outline: none;
+  }
+  ${props => `
```
Comment (mikrotron) [inline packages/Heading/src/Heading.styles.js:60]: Heres' an example fo a block with a bunch of props being used where it totally makes sense to wrap the whole block once in a function, instead of for each line.  And could make it more readable _not_ to destructure props at the beginning, since you'd have a list of 4 or 5.  But I still am not crazy about the way this approach scales down... 🤔

File: packages/Input/src/Input.styles.js [hunk 9]
```diff
@@ -21,122 +21,122 @@ const iconStyles = {
...
-      display: none;
-    }
-  }
+export const InputWrapper = styled.div`
+  ${props => `
+    line-height: 1;
+    position: relative;
 
-  /* Sizes */
+    input.form-input__input {
```
Comment (mikrotron) [inline packages/Input/src/Input.styles.js:29]: You don't need to do it in this PR, it's a bit out of scope, but thought I'd call it out that this "className" approach is a legacy left over from migrating the component from `acl-ui`, and one we'd like to avoid going forward by simply exporting another styled component for this element and applying it in the component.

File: packages/Input/src/Input.styles.js [hunk 10]
```diff
@@ -21,122 +21,122 @@ const iconStyles = {
...
-      box-shadow: ${tokens.highlight.active.noBorder.boxShadow};
-      outline: none;
-    }
-
-    &::-ms-clear {
-      display: none;
-    }
-  }
+export const InputWrapper = styled.div`
+  ${props => `
```
Comment (mikrotron) [inline packages/Input/src/Input.styles.js:25]: Oh, this is probably not the only place, but I think there should be a `css` helper here too to get the IDE syntax highlighting to work properly.

File: packages/ListBox/src/components/Box/Box.styles.js [hunk 11]
```diff
@@ -7,20 +7,16 @@ const isInlineCSS = `
...
-    const widthCSS = props.triggerWidth ? `width: ${props.triggerWidth}px;` : "";
-    return `
-    ${widthCSS}
-    ${inlineCSS}
-    `;
-  }}
+  ${props => `
+    ${props.isInline ? isInlineCSS : ""};
+    ${props.triggerWidth ? `width: ${props.triggerWidth}px;` : ""};
+  `}
```
Comment (mikrotron) [inline packages/ListBox/src/components/Box/Box.styles.js:21]: 👍

File: packages/ListBox/src/components/Label/Label.styles.js [hunk 12]
```diff
@@ -14,6 +14,8 @@ export const labelStyles = `
   width: 100%;
 `;
 
-export const Label = styled.span`
-  ${props => (props.isPlaceholder && !props.hasImplicitAll ? placeholderStyles : "")}
+export const LabelWrapper = styled.span`
+  ${props => `
```
Comment (mikrotron) [inline packages/ListBox/src/components/Label/Label.styles.js:18]: The template string is unnecessary here.

File: packages/ListBox/src/components/Popover/Popover.js [hunk 13]
```diff
@@ -21,9 +21,9 @@ export default function Popover(props) {
   const { children, zIndex } = props;
 
   return (
-    <PopoverStyled {...props} isOpen={state.isOpen} maxWidth={state.triggerWidth} offset={0} zIndex={zIndex}>
+    <sc.PopoverStyled {...props} isOpen={state.isOpen} maxWidth={state.triggerWidth} offset={0} zIndex={zIndex}>
```
Comment (mikrotron) [inline packages/ListBox/src/components/Popover/Popover.js:24]: `<sc.PopoverStyled>` seems redundant to me. `<sc.Popover>` tells me the same thing with fewer words.

File: packages/Pill/src/Pill.styles.js [hunk 14]
```diff
@@ -1,101 +1,126 @@
-import { css } from "styled-components";
+import styled from "styled-components";
 import stylers from "@paprika/stylers";
 import tokens from "@paprika/tokens";
+import RawButton from "@paprika/raw-button";
 
 const pillColorStyles = {
-  black: css`
+  black: `
```
Comment (mikrotron) [inline packages/Pill/src/Pill.styles.js:7]: Probably not a big deal in these cases, you could leave it as is, but the `css` helper was there for that same syntax highlighting reason.

File: packages/Switch/src/Switch.styles.js [hunk 15]
```diff
@@ -8,36 +8,33 @@ const KnobPositionStyles = {
     transform: translateX(12px);
   `,
   medium: css`
-    transform: translate(16px, 0);
+    transform: translate(20px, 0);
   `,
   large: css`
-    transform: translate(20px);
+    transform: translate(28px);
```
Comment (mikrotron) [inline packages/Switch/src/Switch.styles.js:14]: Not sure why these changed... maybe a conflict resolution with integration after you made styling changes to the Switch? Be careful not to undo your previous changes 😬

File: packages/Switch/src/Switch.js [hunk 16]
```diff
@@ -38,8 +37,7 @@ function Switch(props) {
   const { a11yText, canPropagate, isChecked, isDisabled, size, onChange, ...moreProps } = props;
 
   return (
-    <RawButton
-      css={switchStyles}
+    <sc.switchStyles
```
Comment (mikrotron) [inline packages/Switch/src/Switch.js:40]: I think you missed this one... should be `<sc.switchWrapper>` – or, as I'd recommend, just `<sc.Switch>`.

File: packages/Tabs/src/components/List/List.js [hunk 17]
```diff
@@ -37,9 +37,9 @@ export default function List(props) {
   if (a11yText) moreProps["aria-label"] = a11yText;
 
   return (
-    <div {...moreProps} css={tabsListStyles} role="tablist" ref={ref => setTabListRef(ref)} data-pka-anchor="tabs.list">
+    <sc.tabsListStyles {...moreProps} role="tablist" ref={ref => setTabListRef(ref)} data-pka-anchor="tabs.list">
```
Comment (mikrotron) [inline packages/Tabs/src/components/List/List.js:40]: `<sc.TabsList>`

File: packages/Tabs/src/components/Panel/Panel.js [hunk 18]
```diff
@@ -23,9 +23,9 @@ const Panel = props => {
   }
 
   return (
-    <div {...moreProps} css={panelStyles} role="tabpanel" data-pka-anchor="tabs.panel">
+    <sc.panelStyles {...moreProps} role="tabpanel" data-pka-anchor="tabs.panel">
```
Comment (mikrotron) [inline packages/Tabs/src/components/Panel/Panel.js:26]: `<sc.Panel>`

File: packages/Tabs/src/components/Tab/Tab.js [hunk 19]
```diff
@@ -44,26 +43,24 @@ export default function Tab(props) {
 
   if (href && !isDisabled) {
     return (
-      <a
+      <sc.linkStyles
```
Comment (mikrotron) [inline packages/Tabs/src/components/Tab/Tab.js:46]: `<sc.TabLink>`

File: packages/Tabs/src/components/Tab/Tab.js [hunk 20]
```diff
@@ -44,26 +43,24 @@ export default function Tab(props) {
...
       >
         {children}
-      </a>
+      </sc.linkStyles>
     );
   }
 
   return (
-    <RawButton
+    <sc.tabStyles
```
Comment (mikrotron) [inline packages/Tabs/src/components/Tab/Tab.js:60]: `<sc.Tab>`

File: packages/Toast/src/Toast.styles.js [hunk 21]
```diff
@@ -29,7 +29,7 @@ const closeButtonColors = {
   [Kinds.LOCKED]: tokens.color.yellowLighten30,
 };
 
-const fixedStyles = css`
+const fixedStyles = `
```
Comment (mikrotron) [inline packages/Toast/src/Toast.styles.js:32]: Keep the `css`

**Comment by mikrotron** [comment]
Okay, I went through the _whole_ PR this time, and again a few recurring themes:
1. I think you should avoid the `Wrapper` suffix pretty much everywhere. The `sc.` prefix serves a similar purpose of identifying that this is a styled-components wrapper.
2. I think after reading through and thinking about it, we should avoid wrapping the _whole_ styling block in a `props => `...`` function block and instead only wrap the part that relies on `props`.  When that's only one line, probably best to just do it inline as it was before, but if it's two or more, maybe it's a good time to then use a single function wrapper for both lines (or more).
3. Please try to use the `css` helper (before styling blocks wrapped in a template literal).  This helps the IDE recognize that this can be formatted like CSS.  (Not necessary when you use `styled.div` or `styled(SomeComponent)` though.

I think by now you have realized that this was a bit of a trap and that there are hazards to using JavaScript two write CSS – there are _too many ways_ to do the same thing, and establishing conventions and best practices is hard 😅 

But I appreciate your work on this big PR, and maybe you can discuss the conventions that have emerged and any lingering ones that we are still debating on Tuesday during our next UX Dev ShopTalk meeting. Thanks!  🙏

File: packages/ExternalLink/src/ExternalLink.styles.js (hunk 6 above)
Comment (alexzherdev) [reply packages/ExternalLink/src/ExternalLink.styles.js:15]: Does line 7 also need `css` for syntax highlighting?

I'm ambivalent but just sharing some links. Overall this approach seems similar to this, correct? https://twitter.com/NikkitaFTW/status/1262423045874089990
And FWIW I found another one in the comments: https://twitter.com/Saeris/status/1262441528078688256

Comment (mikrotron) [reply packages/ExternalLink/src/ExternalLink.styles.js:15]: I am interested in your ambivalent feelings... or do you really mean _indifferent_?

I actually really like the suggestion from @saeris... so this would look more like:

```jsx
export const ExternalLink = styled.a(props) => css`
  ...
`;
```

or 

```jsx
export const ExternalLink = styled.a({ someProp, someOtherProp }) => css`
  ...
`;
```
```

Also, yes, line 7 _should_ include the `css` helper... I didn't comment on every instance throughout the PR, it comes up for nearly every component.

**Comment by nahumzs** [comment]
This PR will be split in multiple PRs.
